codespan/src/index.rs

```diff
@@ -180,7 +180,18 @@ impl fmt::Display for ColumnOffset {
     }
 }
 
-/// A byte position in a source file
+/// A byte position in a source file. The index is continuous, running over all
+/// files in a `CodeMap`, starting with 1. Zero is reserved for positions that 
+/// do not point to a valid file.
+///
+/// For example, with these files:
+///
+/// - `main.rs` (10 bytes)
+/// - `lib.rs` (314 bytes)
+///
+/// `ByteIndex(0)` points to no file at all, `ByteIndex(2)` points at the
+/// second byte (index 1) in `main.rs` and `ByteIndex(14)` points at the fourth
+/// byte (index 3) in `lib.rs`.
```

Comment:
This is almost correct :). There is also another byte of padding between each file so that we can distinguish the end of one file from the start of another. (I had to double check this).

So `14` ends up pointing at index 2 of `lib.rs`

Reply:
Fixed, hope that was the last error :)